### ISSUE TITLE
Does NOT refresh diffs when update for comments.

### ISSUE THREAD

**Issue by ntsoftabc** [desktop/desktop#7631]
## Version

* GitHub Desktop: 1.6.6

* Operating system: Windows 10 Home Edition

## Steps to Reproduce

1. `Update some file, such as sample.cpp`
2. `Select the updated file and you can see the diffs of the file`
3. `Update comment in file sample.cpp`
4. `Github Desktop does NOT refresh comment in diffs`

**Comment by outofambit** [comment]
Hi @ntsoftabc and thanks for filing this! Just to confirm, this only happens when you update a code comment, not the code itself in the file? Are you switching back and forth between an editor in this process? (We run specific refreshes when the app, is backgrounded and foregrounded.)

A gif showing this behavior would be super helpful if you can! Thank you!

**Comment by ntsoftabc** [comment]
> Hi @ntsoftabc and thanks for filing this! Just to confirm, this only happens when you update a code comment, not the code itself in the file? Are you switching back and forth between an editor in this process? (We run specific refreshes when the app, is backgrounded and foregrounded.)
> 
> A gif showing this behavior would be super helpful if you can! Thank you!

**Comment by ntsoftabc** [comment]
> Hi @ntsoftabc and thanks for filing this! Just to confirm, this only happens when you update a code comment, not the code itself in the file? Are you switching back and forth between an editor in this process? (We run specific refreshes when the app, is backgrounded and foregrounded.)
> 
> A gif showing this behavior would be super helpful if you can! Thank you!

It happens when you update a comment in the code. (At the same time, I also updated some code in the code file) In fact, I did not switch back and forth between an editor in this process. After updated the code and comment, I just save the source file.
Thanks for your reply.

**Comment by shiftkey** [comment]
>  In fact, I did not switch back and forth between an editor in this process. 

I think this might be a key point of the issue. If you can see the diff in Desktop, **but are making changes to the file in another app**, the diff changes will not refresh in Desktop until it receives focus again.

Does that sound like that is happening here @ntsoftabc?

**Comment by ntsoftabc** [comment]
> > In fact, I did not switch back and forth between an editor in this process.
> 
> I think this might be a key point of the issue. If you can see the diff in Desktop, **but are making changes to the file in another app**, the diff changes will not refresh in Desktop until it receives focus again.
> 
> Does that sound like that is happening here @ntsoftabc?

I am afraid what you understand is what I want to ask.  I made changes(both comments and code, at a same file), only differences for code is displayed as diffs, although I also made changes for comment, but the diffs for comments did not display for diffs.
![image](https://user-images.githubusercontent.com/25189166/58425470-6f453100-80cc-11e9-8626-f60c947de901.png)

**Comment by shiftkey** [comment]
@ntsoftabc I've not been able to reproduce this issue myself, and I think it might be down to a specific file and how we handle (or don't handle) some encodings. Are you able to put together a sample repository using a file that shows the problem, so we can reproduce this on our side?

**Comment by ntsoftabc** [comment]
> @ntsoftabc I've not been able to reproduce this issue myself, and I think it might be down to a specific file and how we handle (or don't handle) some encodings. Are you able to put together a sample repository using a file that shows the problem, so we can reproduce this on our side?

I've made a sample repository , you can access it from here:
https://github.com/ntsoftabc/Sample

**Comment by shiftkey** [comment]
@ntsoftabc thanks for the sample repository. I walked through the steps to try and recreate the issue with Desktop `1.7.1-beta1` (the release we're aiming to ship tomorrow) and I couldn't trigger the same behaviour.

Here's what I did:

1. Put the repository back to the state before making any changes:

```shellsession
$ git checkout -b hacking-on-things
Switched to a new branch 'hacking-on-things

$ git reset HEAD~1
HEAD is now at fd9c2eb Before modification

$ git show HEAD
commit fd9c2eb119100320edcbbc6485408449372c2377 (HEAD -> hacking-on-things)
Author: ntsoft <ntsoft@163.com>
Date:   Tue Jun 4 06:21:49 2019 +0800

    Before modification

diff --git a/GithubDesktopBug/sample.c b/GithubDesktopBug/sample.c
index e69de29..df6765e 100644
--- a/GithubDesktopBug/sample.c
+++ b/GithubDesktopBug/sample.c
@@ -0,0 +1,6 @@
+// This is a testxxxxxxxxx
+void main()
+{
+    int a = 1;
+    return;
+}
\ No newline at end of file
```

2. I switched over to Desktop to confirm there are no changes in the working directory:

<img width="691" src="https://user-images.githubusercontent.com/359239/58895265-f21c4c00-86c9-11e9-96a6-561057395598.png">

3. In the editor I made changes to two characters (one in code, one in the comment) to try and trigger the issue. I also saved the file at this point to ensure the changes would show up in Desktop:

<img width="410" src="https://user-images.githubusercontent.com/359239/58895266-f21c4c00-86c9-11e9-831c-f8d33c0eed7a.png">

4. I switched back to Desktop and the one file is marked as changed, with both changes in the diff:

<img width="665" src="https://user-images.githubusercontent.com/359239/58895264-f21c4c00-86c9-11e9-8fce-8d52fe9e3f12.png">

Have I missed anything from your steps here? I'm not sure how to proceed with this one...
